app/src/test/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetControllerTest.kt

```diff
@@ -549,27 +516,6 @@ class BloodPressureEntrySheetControllerTest {
     verify(ui).showDateEntryScreen()
   }
 
-  @Test
-  @Ignore("This test should be isolated and refactored.")
-  @Parameters(method = "params for OpenAs and bp validation errors")
-  fun `when BP entry is active, BP readings are invalid and next arrow is pressed then date entry should not be shown`(
-      openAs: OpenAs,
-      error: BpValidator.Validation
-  ) {
-    whenever(bpValidator.validate(any(), any())).doReturn(error)
-    whenever(bloodPressureRepository.measurement(any())).doReturn(Observable.never())
-
-    uiEvents.run {
-      onNext(SheetCreated(openAs = openAs))
-      onNext(ScreenChanged(BP_ENTRY))
-      onNext(SystolicChanged("-"))
-      onNext(DiastolicChanged("-"))
-      onNext(BloodPressureDateClicked)
-    }
-
-    verify(ui, never()).showDateEntryScreen()
-  }
-
   @Suppress("unused")
   fun `params for OpenAs and bp validation errors`(): List<Any> {
```

Comment:
Did not move this supplier function, because there is another test in the class that relies on this. Hence, this function was copied and later refactored.